packages/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingForm.js

```diff
@@ -0,0 +1,154 @@
+import React, { Fragment } from 'react';
+import gql from 'graphql-tag';
+import { Form } from 'informed';
+import { func, shape, string } from 'prop-types';
+import { useShippingForm } from '@magento/peregrine/lib/talons/CartPage/PriceAdjustments/ShippingMethods/useShippingForm';
+
+import { mergeClasses } from '../../../../classify';
+import { isRequired } from '../../../../util/formValidators';
+import Button from '../../../Button';
+import Field from '../../../Field';
+import Select from '../../../Select';
+import TextInput from '../../../TextInput';
+import defaultClasses from './shippingForm.css';
+import { ShippingMethodsFragment } from './shippingMethodsFragments';
+import { GET_SHIPPING_METHODS } from './shippingMethods';
+
+const ShippingForm = props => {
+    const { hasMethods, selectedShippingFields } = props;
+
+    const {
+        countries,
+        handleCountryChange,
+        handleOnSubmit,
+        handleZipChange,
+        isCountriesLoading,
+        setShippingLoading,
+        states
+    } = useShippingForm({
+        getCountriesQuery: GET_COUNTRIES_QUERY,
+        getStatesQuery: GET_STATES_QUERY,
+        selectedValues: selectedShippingFields,
+        setShippingMutation: SET_SHIPPING_MUTATION,
+        shippingMethodsQuery: GET_SHIPPING_METHODS
+    });
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const stateProps = {
+        field: 'state',
+        validate: isRequired
+    };
+    const stateField = states.length ? (
+        <Select {...stateProps} items={states} />
+    ) : (
+        <TextInput {...stateProps} />
+    );
+
+    return (
+        <Fragment>
+            <h3 className={classes.formTitle}>Destination</h3>
+            <Form
+                className={classes.root}
+                initialValues={selectedShippingFields}
+                onSubmit={handleOnSubmit}
+            >
+                <Field
+                    id="country"
+                    label="Country"
+                    classes={{ root: classes.country }}
+                >
+                    <Select
+                        disabled={isCountriesLoading ? true : null}
```

Comment:
Can this just be set to `isCountriesLoading`?

Reply:
I thought there was some React magic where null props aren't rendered to the DOM, which I thought was preferable. I seem to be the only one using this pattern though, so I may go ahead and make this change. Second commit inbound.

Reply:
Fixed in f9a3c8e